### PR TITLE
[Form] Clean unused code

### PR DESCRIPTION
**Pull request by alamirault**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Issues        | Fix #... 
| License       | MIT

Remove unused code

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php [hunk 1]
```diff
@@ -192,44 +190,40 @@ protected function getNumberFormatter(): \NumberFormatter
...
-                    $number = round($number, 0, \PHP_ROUND_HALF_DOWN);
-                    break;
-            }
-
-            $number = 1 === $roundingCoef ? (int) $number : $number / $roundingCoef;
-        }
+        // shift number to maintain the correct scale during rounding
+        $roundingCoef = 10 ** $this->scale;
 
-        return $number;
```
Comment (yceruto) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php:233]: you've missed this line at the end

**Review by yceruto — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php (hunk 1 above)
Comment (alamirault) [reply src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php:233]: Oops, fixed

**Review by yceruto — APPROVED** [review]

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php [hunk 2]
```diff
@@ -192,43 +190,41 @@ protected function getNumberFormatter(): \NumberFormatter
...
-            $number = 1 === $roundingCoef ? (int) $number : $number / $roundingCoef;
-        }
+        if (self::FRACTIONAL == $this->type) {
+            $roundingCoef *= 100;
+        }
+
+        // string representation to avoid rounding errors, similar to bcmul()
+        $number = (string) ($number * $roundingCoef);
+
+        switch ($this->roundingMode) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php]: let's use a match statement for this one

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php [hunk 3]
```diff
@@ -192,43 +190,41 @@ protected function getNumberFormatter(): \NumberFormatter
...
-                    $number = round($number, 0, \PHP_ROUND_HALF_UP);
-                    break;
-                case \NumberFormatter::ROUND_HALFDOWN:
-                    $number = round($number, 0, \PHP_ROUND_HALF_DOWN);
-                    break;
-            }
-
-            $number = 1 === $roundingCoef ? (int) $number : $number / $roundingCoef;
-        }
+        if (self::FRACTIONAL == $this->type) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php]:
```suggestion
        if (self::FRACTIONAL === $this->type) {
```

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php (hunk 2 above)
Comment (alamirault) [reply src/Symfony/Component/Form/Extension/Core/DataTransformer/PercentToLocalizedStringTransformer.php]: I did same on `NumberToLocalizedStringTransformer`

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @alamirault.

**Pull request merged**